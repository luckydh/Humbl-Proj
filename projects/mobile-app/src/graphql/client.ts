import { ApolloClient, createHttpLink, InMemoryCache, from, ApolloLink } from "@apollo/client";
import { setContext } from "@apollo/client/link/context";
import { onError } from "@apollo/client/link/error";
import "firebase/auth";
import { Storage } from "@capacitor/storage";
import { trackEvent } from "utils/analytics/Segment";
import EVENTS from "utils/analytics/AnalyticEvents";
import { setMinAppVersion, currAppVersion, minAppVersionVar as currMinAppVersion } from "utils/update/update";
import { captureException } from "ErrorLogger";
import { GraphQLError } from "graphql";
import firebase from "../Firebase";

const httpLink = createHttpLink({
  uri: process.env.REACT_APP_HUMBL_DATA_URL,
  // credentials: 'same-origin',
});

const error = onError(({ operation, networkError, graphQLErrors, response }) => {
  trackEvent(EVENTS.API_ERROR, {
    operation,
    networkError,
    graphQLErrors,
  });
  if (networkError) {
    captureException(networkError);
  }

  if (graphQLErrors) {
    graphQLErrors.forEach((e) => {
      const sentryError = new Error();
      const name = e.extensions?.code ?? e.message;
      sentryError.name = name;
      sentryError.message = e.message;
      captureException(sentryError, generateGraphQLErrorMeta(e));
    });
  }

  if (response?.errors) {
    // eslint-disable-next-line no-param-reassign
    response.errors = response.errors.map((err) => ({
      ...err,
      humblErrorCode: err.extensions?.code,
    }));
  }
});

function generateGraphQLErrorMeta(graphqlError: GraphQLError) {
  return Object.entries(graphqlError).reduce((curr, next) => {
    const [k, v] = next;
    // Ignore message since we use that as the main error text anyways.
    if (k !== "message") {
      try {
        curr[k] = JSON.stringify(v);
      } catch (e) {
        captureException(e);
      }
    }

    return curr;
  }, {} as Record<string, string>);
}

const authLink = setContext(async (_, { headers }) => {
  // get the authentication token from firebase if it exists
  const token = await firebase.auth().currentUser?.getIdToken();
  // return the headers to the context so httpLink can read them
  Storage.migrate();
  const accountId = await Storage.get({ key: "defaultAccountId" });
  const headerObject = {
    ...headers,
    authorization: token ? `Bearer ${token}` : "",
    version: currAppVersion,
  };
  // If we have a default account in local storage to work with, send it along
  if (accountId.value) {
    headerObject.account = accountId.value;
  }
  return {
    headers: headerObject,
  };
});

const responseLink = new ApolloLink((operation, forward) => forward(operation).map((response) => {
    const context = operation.getContext();
    const minAppVersion: string = context?.response?.headers?.get("minimum-app-version") ?? "";

    if (minAppVersion !== currMinAppVersion()) {
      setMinAppVersion(minAppVersion);
    }

    return response;
  }));

const client = new ApolloClient({
  cache: new InMemoryCache(),
  uri: process.env.REACT_APP_HUMBL_DATA_URL,
  link: from([error, authLink, responseLink.concat(httpLink)]),
  defaultOptions: { query: { fetchPolicy: "cache-first" } },
});

export default client;
