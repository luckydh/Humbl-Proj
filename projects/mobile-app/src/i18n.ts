import i18n from "i18next";
import { initReactI18next, useTranslation } from "react-i18next";
import { register } from "timeago.js";
import { es, pt_BR } from "timeago.js/lib/lang";
import HttpApi from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";
//import resources from "./locales/index";
// resources: resources,

export type TFnType = ReturnType<typeof useTranslation>["t"];

register("es", es);
register("pt", pt_BR);

i18n
  .use(HttpApi)
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    backend: {
      loadPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/{{lng}}/{{ns}}",
      addPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/add/{{lng}}/{{ns}}",
      allowMultiLoading: false,
    },
    //resources,
    //If we don't have a transalation and only have empty string, display fallback
    returnEmptyString: false,
    fallbackLng: "en",
    debug: false,
    ns: "translation",
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
