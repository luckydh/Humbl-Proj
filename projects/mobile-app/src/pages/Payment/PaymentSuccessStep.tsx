import React from "react";
import { AccountType } from "generated/graphql";
import { Button } from "components/Button/Button";
import { useTranslation } from "react-i18next";
import { formatCurrency, getCurrencySymbol } from "utils/currency";
import { ProfileLayout } from "components/PageTemplates/ProfileLayout";
import moment from "moment";
import CheckMark from "components/CheckMark";
import { useHistory } from "react-router";
import { useCurrentAccountHomePath } from "hooks/useCurrentAccountHomePath";

interface Props {
  account: AccountType;
  totalAmount?: number;
  currency: string;
  onClickReview: () => void;
}

export const PaymentSuccessStep: React.FC<Props> = ({ account, totalAmount, currency, onClickReview }) => {
  const { t } = useTranslation();
  const history = useHistory();
  const homePath = useCurrentAccountHomePath();
  return (
    <ProfileLayout showBackButton={false}>
      <div className="flex flex-col pt-14 h-full justify-between">
        <div className=" flex items-center flex-col relative w-full">
          <CheckMark className="" noBackground />
        </div>
        <div className="flex-1 flex flex-col flex-grow justify-around text-center text-white pb-6">
          <div>
            <h1 className="text-3xl">{t("payment-page.action.payment-success")}</h1>
            <div className="text-base text-white-faded">
              {t("payment-page.action.at", { date: moment().format("MMM D YYYY"), time: moment().format("hh:MM A") })}
            </div>
          </div>
          <div>
            <h1 className="text-2xl">{account.displayName}</h1>
            <h1 className="text-3xl">
              {getCurrencySymbol(currency)}
              {formatCurrency(totalAmount!)} {currency}
            </h1>
          </div>
        </div>
        <div className="relative bottom-0 flex py-4 flex-col">
          <Button className="mb-2" onClick={onClickReview}>
            {t("payment-page.action.write-a-review")}
          </Button>
          <Button
            variant="custom"
            className="rounded-md font-bold w-full border-blue-dark text-blue-dark py-2 px-4 border-2 border-solid"
            onClick={() => history.push(homePath)}
          >
            {t("payment-page.action.go-back-home")}
          </Button>
        </div>
      </div>
    </ProfileLayout>
  );
};

export default PaymentSuccessStep;
