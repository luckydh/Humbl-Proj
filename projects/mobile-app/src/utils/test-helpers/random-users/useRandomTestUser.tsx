import { AccountType } from "generated/graphql";
import randomUsers from "../randomUsers.json";

export interface GeneratedUserRecord {
  cell: string;
  dob: { date: string; age: number };
  gender: string;
  id: { name: string; value: string | null };
  location: {
    city: string;
    street: { number: number; name: string };
    coordinates: { latitude: string; longitude: string };
    country: string;
    postcode: string | number;
    state: string;
    timezone: {
      description: string;
      offset: string;
    };
  };
  login: {
    md5: string;
    password: string;
    salt: string;
    sha1: string;
    sha256: string;
    username: string;
    uuid: string;
  };
  name: {
    first: string;
    last: string;
    title: string;
  };
  nat: string;
  phone: string;
  picture: {
    large: string;
    medium: string;
    thumbnail: string;
  };
  registered: {
    age: number;
    date: string;
  };
}
export interface GeneratedUser {
  chosenUser: number;
  user: GeneratedUserRecord;
}

export const useRandomTestUser = (userArrayPosition?: number) => {
  const randomPosition = userArrayPosition || Math.floor(Math.random() * randomUsers.results.length);
  const user = randomUsers.results[randomPosition];
  return { chosenUser: randomPosition, user };
};

export const getRandomTestUser = (userArrayPosition?: number) => {
  const randomPosition = userArrayPosition || Math.floor(Math.random() * randomUsers.results.length);
  const user = randomUsers.results[randomPosition];
  return { chosenUser: randomPosition, user };
};

export const getTestUsers = (numberOfUsers: number, startPosition?: number) =>
  Array.from({ length: numberOfUsers }, () => getRandomTestUser(startPosition));

export const convertGeneratedtoAccounts = (users: GeneratedUser[]): AccountType[] =>
  users.map((currentUser: GeneratedUser) => ({
    id: currentUser.user.login.uuid,
    userName: currentUser.user.login.username,
    displayName: `${currentUser.user.name.first} ${currentUser.user.name.last}`,
    image: currentUser.user.picture.thumbnail,
    isMerchant: false,
  }));
