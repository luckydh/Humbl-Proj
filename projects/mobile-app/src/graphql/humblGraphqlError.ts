import { ApolloError } from "@apollo/client";
import { GraphQLError } from "graphql";
import i18n from "i18n";

export type HumblErrorCode = keyof typeof errorCodesTranslationKeys;

export interface HumblError {
  humblErrorCode: HumblErrorCode;
}

export interface HumblGraphQLError extends GraphQLError {
  humblErrorCode?: HumblErrorCode;
}

const errorCodesTranslationKeys = {
  SERVER_ERROR: "crypto-wallet.buy.order-preview.error-modal.fallback-message",
  IDENTITY_MISMATCH: "crypto-wallet.buy.order-preview.error-modal.identity-mismatch",
  PHONE_MISMATCH_INVALID: "crypto-wallet.buy.order-preview.error-modal.phone-mismatch-invalid",
  TOO_MUCH_ABUSE_SUSPICIOUS: "crypto-wallet.buy.order-preview.error-modal.abuse-suspicious",
  AUTHORIZATION_FAILED: "crypto-wallet.buy.order-preview.error-modal.auth-failed",
  ORDER_EXPIRED: "crypto-wallet.buy.order-preview.error-modal.order-expired",
  DECLINED_DO_NOT_HONOR: "crypto-wallet.buy.order-preview.error-modal.declined-do-not-honor",
  INSUFFICIENT_FUNDS: "crypto-wallet.buy.order-preview.error-modal.insufficient-funds",
  RESTRICTED: "crypto-wallet.buy.order-preview.error-modal.restricted",
  CARD_ISSUER_DECLINED: "crypto-wallet.buy.order-preview.error-modal.card-issuer-declined",
  LIMIT_REACHED: "crypto-wallet.buy.order-preview.error-modal.limit-reached",
  VALIDATION_ERROR: "crypto-wallet.buy.order-preview.error-modal.validation-error",
};

const fallback = errorCodesTranslationKeys.SERVER_ERROR;

export function getPotentialHumblError(graphQLErrors: ReadonlyArray<HumblGraphQLError>): HumblError | undefined {
  // This can sometimes trigger with 0 graphQLErrors hence this guard.
  const humblErrorCode = graphQLErrors[0]?.humblErrorCode;
  return humblErrorCode && { humblErrorCode };
}

export function getErrorMessage(error?: ApolloError) {
  if (!error) {
    return;
  }
  const humblError = getPotentialHumblError(error.graphQLErrors);
  return i18n.t(mapErrorCodeToTranslationKey(humblError?.humblErrorCode));
}

export function mapErrorCodeToTranslationKey(errorCode: HumblErrorCode = "SERVER_ERROR") {
  return errorCodesTranslationKeys[errorCode] ?? fallback;
}
