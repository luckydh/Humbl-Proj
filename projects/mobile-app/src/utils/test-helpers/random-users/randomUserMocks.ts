import { getRandomTestUser, useRandomTestUser } from "./useRandomTestUser";
import createApolloMock from "../../../generated/createApolloMock";
import {
  GetAccountByIdDocument,
  MeDocument,
  MyAccountsDocument,
  MyUserProfileDocument,
} from "../../../generated/graphql";

export const useGetAccountByIdQueryMock = (userArrayPosition?: number, options?: Record<string, any>) => {
  const { user } = useRandomTestUser(userArrayPosition);

  return createApolloMock(
    GetAccountByIdDocument,
    { id: `${userArrayPosition}` },
    {
      data: {
        accountById: {
          userName: user.login.username,
          hasMultipleAccounts: false,
          hasMerchantAccount: false,
          isMerchant: false,
          displayName: user.login.username,
          image: user.picture.large,
          id: user.login.uuid,
          city: user.location.city,
          isPrivate: false,
          phone: user.phone,
          country: {
            name: user.location.country,
            alpha2: user.nat.toLowerCase(),
          },
          ...options,
        },
      },
    },
    { addTypename: false }
  );
};

export const useMyUserProfileQueryMock = (userArrayPosition?: number) => {
  const { user } = useRandomTestUser(userArrayPosition);
  return createApolloMock(
    MyUserProfileDocument,
    {},
    {
      data: {
        myUserProfile: {
          firstName: user.name.first,
          lastName: user.name.last,
          fullName: `${user.name.first} ${user.name.last}`,
          createdOn: new Date(),
        },
      },
    },
    { addTypename: false }
  );
};

export const useMeQueryMock = (userArrayPosition?: number) => {
  const { user } = useRandomTestUser(userArrayPosition);
  return createApolloMock(
    MeDocument,
    {},
    {
      data: {
        me: {
          userName: user.login.username,
          hasMultipleAccounts: false,
          hasMerchantAccount: false,
          isMerchant: false,
          displayName: user.login.username,
          image: user.picture.large,
          id: user.login.uuid,
          city: user.location.city,
          isPrivate: false,
          phone: user.phone,
          country: {
            name: user.location.country,
            alpha2: user.nat.toLowerCase(),
          },
        },
      },
    },
    { addTypename: false }
  );
};

export const useMyAccountQueryMock = (userPositions: Array<number>) => {
  const accounts = userPositions
    .map((position) => getRandomTestUser(position))
    .map(({ user }) => ({
      image: user.picture.large,
      displayName: user.login.username,
      userName: user.login.username,
      id: user.login.uuid,
    }));
  return createApolloMock(
    MyAccountsDocument,
    {},
    {
      data: {
        accounts,
      },
    },
    { addTypename: false }
  );
};
