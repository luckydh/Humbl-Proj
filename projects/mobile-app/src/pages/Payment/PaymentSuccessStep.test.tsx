import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { advanceTo, clear } from "jest-date-mock";
import { StaticRouter, MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import { useGetAccountByIdQueryMock } from "utils/test-helpers/random-users/randomUserMocks";
import { AccountType } from "generated/graphql";
import i18n from "../../i18n";

import PaymentSuccessStep from "./PaymentSuccessStep";

if (i18n.options) {
  i18n.options.react = {
    useSuspense: false,
  };
}

const testAccount: AccountType = {
  id: "test-12345",
  image: "",
  userName: "test",
  displayName: "Test",
};

const defaultProps = {
  currency: "USD",
  account: testAccount,
  totalAmount: 10000,
  onClickReview: jest.fn(),
};

beforeAll(() => {
  advanceTo(new Date(2021, 0, 1, 0, 0, 0, 0));
});

afterAll(() => {
  clear();
});

describe("PaymentSuccessStep", () => {
  it("should display the total amount with correct currency symbol", () => {
    const Providers: React.FC = ({ children }) => (
      <MockedProvider mocks={[useGetAccountByIdQueryMock(1)]} addTypename={false}>
        <MemoryRouter>{children}</MemoryRouter>
      </MockedProvider>
    );

    const { getByText, asFragment, rerender } = render(<PaymentSuccessStep {...defaultProps} currency="USD" />, {
      wrapper: Providers,
    });

    expect(getByText("$100.00 USD")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();

    rerender(<PaymentSuccessStep {...defaultProps} currency="MXN" />);

    expect(getByText("$100.00 MXN")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();

    rerender(<PaymentSuccessStep {...defaultProps} currency="CAD" />);

    expect(getByText("$100.00 CAD")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call onClickReview when clicking the CTA", async () => {
    const onClickReview = jest.fn();

    const { getByText } = render(
      <StaticRouter>
        <MockedProvider mocks={[useGetAccountByIdQueryMock(1)]} addTypename={false}>
          <PaymentSuccessStep {...defaultProps} onClickReview={onClickReview} />
        </MockedProvider>
      </StaticRouter>
    );

    userEvent.click(getByText("payment-page.action.write-a-review"));

    expect(onClickReview).toBeCalled();
  });
});
