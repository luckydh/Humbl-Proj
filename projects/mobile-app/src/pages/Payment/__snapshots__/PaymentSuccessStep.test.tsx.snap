// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSuccessStep should display the total amount with correct currency symbol 1`] = `
<DocumentFragment>
  <div
    class="ion-page bg-profiles safe-area-top"
  >
    <div
      class="relative flex-1 flex flex-col justify-between w-full h-full"
    >
      <div
        class="flex items-center justify-center py-6 flex-shink-0"
      >
        <svg
          aria-label="HUMBL_LOGO"
          class="inline-block"
          height="31"
          width="143"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#FFF"
            fill-rule="nonzero"
          >
            <path
              d="M18.418 1v11.484H7.151V1H0v29.794h7.151V18.335h11.267v12.459h7.259V1zM37.594 1v17.118a7.15 7.15 0 001.3 4.55 4.659 4.659 0 003.9 1.517 5.092 5.092 0 003.9-1.517 6.934 6.934 0 001.3-4.55V1h7.151v17.118c.09 2.45-.47 4.88-1.625 7.042a11.376 11.376 0 01-4.442 4.334 13.543 13.543 0 01-6.392 1.406c-2.18.06-4.339-.423-6.284-1.408a10.4 10.4 0 01-4.334-4.334 14.518 14.518 0 01-1.517-7.148V1h7.043zM93.39 1.217v29.577h-7.151V12.376l-6.392 18.527h-6.175L67.28 12.376v18.527h-7.151V1.217h8.776l8.017 20.585 7.907-20.585zM120.475 18.118a6.934 6.934 0 011.625 4.55 7.476 7.476 0 01-2.492 5.959 11.159 11.159 0 01-7.259 2.167H98.482V.998h13.543c2.5-.159 4.98.528 7.042 1.95a6.934 6.934 0 012.492 5.634 6.609 6.609 0 01-1.408 4.442 6.934 6.934 0 01-3.792 2.384 8.884 8.884 0 014.117 2.709m-14.952-4.875h4.767c2.492 0 3.792-1.083 3.792-3.25s-1.3-3.25-3.792-3.25h-4.658l-.109 6.5zm9.209 8.451a2.817 2.817 0 00-1.083-2.492 4.225 4.225 0 00-3.034-.867h-5.092v6.609h5.092c2.709 0 4.117-1.083 4.117-3.25M133.044 25.268h9.642v5.525h-16.793V1h7.151zM140.519 1a2.167 2.167 0 110 4.334 2.167 2.167 0 010-4.334m0 4.117a1.95 1.95 0 10-1.95-1.95 1.95 1.95 0 001.95 1.95m-.758-3.25h.975c.542 0 .867.217.867.758a.65.65 0 01-.65.65l.758 1.192h-.217l-.758-1.192h-.65v1.192h-.325v-2.6zm.217 1.192h.65c.433 0 .65-.108.65-.433s-.217-.542-.65-.542h-.65v.975z"
            />
          </g>
        </svg>
      </div>
      <div
        class="flex flex-col flex-1 overflow-y-auto ease-in-out"
        id="keyboardAwareView"
      >
        <div
          class="px-6 flex-grow"
        >
          <div
            class="flex flex-col pt-14 h-full justify-between"
          >
            <div
              class=" flex items-center flex-col relative w-full"
            >
              <div
                class="z-50 border-white p-12 rounded-full opacity-95"
                data-testid="checkmark"
                style="top: 0px; width: 219px; height: 219px; border-width: 7px;"
              >
                <svg
                  class="w-full h-full"
                  fill="none"
                  height="87"
                  viewBox="0 0 103 87"
                  width="103"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M29.668 86.514h-.053a.608.608 0 01-.452-.194l-.038-.042L.773 58.28a.52.52 0 01-.136-.386l.009-.155-.012-.154a.62.62 0 01.176-.482l9.197-9.099.03-.033a.682.682 0 01.972-.032l.096.096 18.63 18.17L91.948.702a.688.688 0 01.978-.032l.082.084 9.21 9.113a.58.58 0 01.149.44l-.012.155.009.155a.525.525 0 01-.154.4L30.255 86.292a.707.707 0 01-.512.224l-.075-.002z"
                    fill="#fff"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex-1 flex flex-col flex-grow justify-around text-center text-white pb-6"
            >
              <div>
                <h1
                  class="text-3xl"
                >
                  payment-page.action.payment-success
                </h1>
                <div
                  class="text-base text-white-faded"
                >
                  payment-page.action.at
                </div>
              </div>
              <div>
                <h1
                  class="text-2xl"
                >
                  Test
                </h1>
                <h1
                  class="text-3xl"
                >
                  $100.00 USD
                </h1>
              </div>
            </div>
            <div
              class="relative bottom-0 flex py-4 flex-col"
            >
              <button
                class="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none mb-2"
                style="min-width: 160px;"
                type="button"
              >
                payment-page.action.write-a-review
              </button>
              <button
                class="text-white select-none outline-none flex justify-center items-center py-3 px-6 text-md sm:leading-none rounded-md font-bold w-full border-blue-dark text-blue-dark py-2 px-4 border-2 border-solid"
                type="button"
              >
                payment-page.action.go-back-home
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PaymentSuccessStep should display the total amount with correct currency symbol 2`] = `
<DocumentFragment>
  <div
    class="ion-page bg-profiles safe-area-top"
  >
    <div
      class="relative flex-1 flex flex-col justify-between w-full h-full"
    >
      <div
        class="flex items-center justify-center py-6 flex-shink-0"
      >
        <svg
          aria-label="HUMBL_LOGO"
          class="inline-block"
          height="31"
          width="143"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#FFF"
            fill-rule="nonzero"
          >
            <path
              d="M18.418 1v11.484H7.151V1H0v29.794h7.151V18.335h11.267v12.459h7.259V1zM37.594 1v17.118a7.15 7.15 0 001.3 4.55 4.659 4.659 0 003.9 1.517 5.092 5.092 0 003.9-1.517 6.934 6.934 0 001.3-4.55V1h7.151v17.118c.09 2.45-.47 4.88-1.625 7.042a11.376 11.376 0 01-4.442 4.334 13.543 13.543 0 01-6.392 1.406c-2.18.06-4.339-.423-6.284-1.408a10.4 10.4 0 01-4.334-4.334 14.518 14.518 0 01-1.517-7.148V1h7.043zM93.39 1.217v29.577h-7.151V12.376l-6.392 18.527h-6.175L67.28 12.376v18.527h-7.151V1.217h8.776l8.017 20.585 7.907-20.585zM120.475 18.118a6.934 6.934 0 011.625 4.55 7.476 7.476 0 01-2.492 5.959 11.159 11.159 0 01-7.259 2.167H98.482V.998h13.543c2.5-.159 4.98.528 7.042 1.95a6.934 6.934 0 012.492 5.634 6.609 6.609 0 01-1.408 4.442 6.934 6.934 0 01-3.792 2.384 8.884 8.884 0 014.117 2.709m-14.952-4.875h4.767c2.492 0 3.792-1.083 3.792-3.25s-1.3-3.25-3.792-3.25h-4.658l-.109 6.5zm9.209 8.451a2.817 2.817 0 00-1.083-2.492 4.225 4.225 0 00-3.034-.867h-5.092v6.609h5.092c2.709 0 4.117-1.083 4.117-3.25M133.044 25.268h9.642v5.525h-16.793V1h7.151zM140.519 1a2.167 2.167 0 110 4.334 2.167 2.167 0 010-4.334m0 4.117a1.95 1.95 0 10-1.95-1.95 1.95 1.95 0 001.95 1.95m-.758-3.25h.975c.542 0 .867.217.867.758a.65.65 0 01-.65.65l.758 1.192h-.217l-.758-1.192h-.65v1.192h-.325v-2.6zm.217 1.192h.65c.433 0 .65-.108.65-.433s-.217-.542-.65-.542h-.65v.975z"
            />
          </g>
        </svg>
      </div>
      <div
        class="flex flex-col flex-1 overflow-y-auto ease-in-out"
        id="keyboardAwareView"
      >
        <div
          class="px-6 flex-grow"
        >
          <div
            class="flex flex-col pt-14 h-full justify-between"
          >
            <div
              class=" flex items-center flex-col relative w-full"
            >
              <div
                class="z-50 border-white p-12 rounded-full opacity-95"
                data-testid="checkmark"
                style="top: 0px; width: 219px; height: 219px; border-width: 7px;"
              >
                <svg
                  class="w-full h-full"
                  fill="none"
                  height="87"
                  viewBox="0 0 103 87"
                  width="103"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M29.668 86.514h-.053a.608.608 0 01-.452-.194l-.038-.042L.773 58.28a.52.52 0 01-.136-.386l.009-.155-.012-.154a.62.62 0 01.176-.482l9.197-9.099.03-.033a.682.682 0 01.972-.032l.096.096 18.63 18.17L91.948.702a.688.688 0 01.978-.032l.082.084 9.21 9.113a.58.58 0 01.149.44l-.012.155.009.155a.525.525 0 01-.154.4L30.255 86.292a.707.707 0 01-.512.224l-.075-.002z"
                    fill="#fff"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex-1 flex flex-col flex-grow justify-around text-center text-white pb-6"
            >
              <div>
                <h1
                  class="text-3xl"
                >
                  payment-page.action.payment-success
                </h1>
                <div
                  class="text-base text-white-faded"
                >
                  payment-page.action.at
                </div>
              </div>
              <div>
                <h1
                  class="text-2xl"
                >
                  Test
                </h1>
                <h1
                  class="text-3xl"
                >
                  $100.00 MXN
                </h1>
              </div>
            </div>
            <div
              class="relative bottom-0 flex py-4 flex-col"
            >
              <button
                class="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none mb-2"
                style="min-width: 160px;"
                type="button"
              >
                payment-page.action.write-a-review
              </button>
              <button
                class="text-white select-none outline-none flex justify-center items-center py-3 px-6 text-md sm:leading-none rounded-md font-bold w-full border-blue-dark text-blue-dark py-2 px-4 border-2 border-solid"
                type="button"
              >
                payment-page.action.go-back-home
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PaymentSuccessStep should display the total amount with correct currency symbol 3`] = `
<DocumentFragment>
  <div
    class="ion-page bg-profiles safe-area-top"
  >
    <div
      class="relative flex-1 flex flex-col justify-between w-full h-full"
    >
      <div
        class="flex items-center justify-center py-6 flex-shink-0"
      >
        <svg
          aria-label="HUMBL_LOGO"
          class="inline-block"
          height="31"
          width="143"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#FFF"
            fill-rule="nonzero"
          >
            <path
              d="M18.418 1v11.484H7.151V1H0v29.794h7.151V18.335h11.267v12.459h7.259V1zM37.594 1v17.118a7.15 7.15 0 001.3 4.55 4.659 4.659 0 003.9 1.517 5.092 5.092 0 003.9-1.517 6.934 6.934 0 001.3-4.55V1h7.151v17.118c.09 2.45-.47 4.88-1.625 7.042a11.376 11.376 0 01-4.442 4.334 13.543 13.543 0 01-6.392 1.406c-2.18.06-4.339-.423-6.284-1.408a10.4 10.4 0 01-4.334-4.334 14.518 14.518 0 01-1.517-7.148V1h7.043zM93.39 1.217v29.577h-7.151V12.376l-6.392 18.527h-6.175L67.28 12.376v18.527h-7.151V1.217h8.776l8.017 20.585 7.907-20.585zM120.475 18.118a6.934 6.934 0 011.625 4.55 7.476 7.476 0 01-2.492 5.959 11.159 11.159 0 01-7.259 2.167H98.482V.998h13.543c2.5-.159 4.98.528 7.042 1.95a6.934 6.934 0 012.492 5.634 6.609 6.609 0 01-1.408 4.442 6.934 6.934 0 01-3.792 2.384 8.884 8.884 0 014.117 2.709m-14.952-4.875h4.767c2.492 0 3.792-1.083 3.792-3.25s-1.3-3.25-3.792-3.25h-4.658l-.109 6.5zm9.209 8.451a2.817 2.817 0 00-1.083-2.492 4.225 4.225 0 00-3.034-.867h-5.092v6.609h5.092c2.709 0 4.117-1.083 4.117-3.25M133.044 25.268h9.642v5.525h-16.793V1h7.151zM140.519 1a2.167 2.167 0 110 4.334 2.167 2.167 0 010-4.334m0 4.117a1.95 1.95 0 10-1.95-1.95 1.95 1.95 0 001.95 1.95m-.758-3.25h.975c.542 0 .867.217.867.758a.65.65 0 01-.65.65l.758 1.192h-.217l-.758-1.192h-.65v1.192h-.325v-2.6zm.217 1.192h.65c.433 0 .65-.108.65-.433s-.217-.542-.65-.542h-.65v.975z"
            />
          </g>
        </svg>
      </div>
      <div
        class="flex flex-col flex-1 overflow-y-auto ease-in-out"
        id="keyboardAwareView"
      >
        <div
          class="px-6 flex-grow"
        >
          <div
            class="flex flex-col pt-14 h-full justify-between"
          >
            <div
              class=" flex items-center flex-col relative w-full"
            >
              <div
                class="z-50 border-white p-12 rounded-full opacity-95"
                data-testid="checkmark"
                style="top: 0px; width: 219px; height: 219px; border-width: 7px;"
              >
                <svg
                  class="w-full h-full"
                  fill="none"
                  height="87"
                  viewBox="0 0 103 87"
                  width="103"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M29.668 86.514h-.053a.608.608 0 01-.452-.194l-.038-.042L.773 58.28a.52.52 0 01-.136-.386l.009-.155-.012-.154a.62.62 0 01.176-.482l9.197-9.099.03-.033a.682.682 0 01.972-.032l.096.096 18.63 18.17L91.948.702a.688.688 0 01.978-.032l.082.084 9.21 9.113a.58.58 0 01.149.44l-.012.155.009.155a.525.525 0 01-.154.4L30.255 86.292a.707.707 0 01-.512.224l-.075-.002z"
                    fill="#fff"
                  />
                </svg>
              </div>
            </div>
            <div
              class="flex-1 flex flex-col flex-grow justify-around text-center text-white pb-6"
            >
              <div>
                <h1
                  class="text-3xl"
                >
                  payment-page.action.payment-success
                </h1>
                <div
                  class="text-base text-white-faded"
                >
                  payment-page.action.at
                </div>
              </div>
              <div>
                <h1
                  class="text-2xl"
                >
                  Test
                </h1>
                <h1
                  class="text-3xl"
                >
                  $100.00 CAD
                </h1>
              </div>
            </div>
            <div
              class="relative bottom-0 flex py-4 flex-col"
            >
              <button
                class="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none mb-2"
                style="min-width: 160px;"
                type="button"
              >
                payment-page.action.write-a-review
              </button>
              <button
                class="text-white select-none outline-none flex justify-center items-center py-3 px-6 text-md sm:leading-none rounded-md font-bold w-full border-blue-dark text-blue-dark py-2 px-4 border-2 border-solid"
                type="button"
              >
                payment-page.action.go-back-home
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
